cmake/checks/check_01_cxx_features.cmake

```diff
@@ -501,6 +502,16 @@ OPTION(DEAL_II_WITH_CXX17
 LIST(APPEND DEAL_II_FEATURES CXX17)
 SET(FEATURE_CXX17_PROCESSED TRUE)
 
+#
+# The macro DEAL_II_CONSTEXPR allows using c++ constexpr features in a portable way
+#
+IF (DEAL_II_WITH_CXX14)
+  MESSAGE(STATUS "Enabling CXX14 constexpr features")
+  SET(DEAL_II_CONSTEXPR "constexpr")
+ELSE()
+  SET(DEAL_II_CONSTEXPR " ")
```

Comment:
If you replace `inline` with `constexpr` you are effectively removing `inline` when compiling without `C++14` support. That is likely not intended.

Reply:
The functions I decorate with `constexpr` are one-liners that will be constexpr even in old c++11 compilers.

All other functions are decorated with `DEAL_II_CONSTEXPR inline` which will remain inline both with or without c++14 support. Some of these functions are additionally decorated with `DEAL_II_ALWAYS_INLINE`.

So there are no portability issues with regard to the definition of `DEAL_II_CONSTEXPR`.